src/Expander/TC.hs

```diff
@@ -97,8 +99,8 @@ freshMeta = do
   return ptr
 
 inst :: Scheme Ty -> [Ty] -> Expand Ty
-inst (Scheme n ty) ts
-  | length ts /= fromIntegral n =
+inst (Scheme argKinds ty) ts
+  | length ts /= length argKinds =
```

Comment:
currently, type arguments always have kind `*`, so it is not necessary to check that each element of `ts` has the right kind.

Reply:
This seems like the kind of thing that will be easy to forget later...

Reply:
Fixed in #92 .